### PR TITLE
Add usi support to strategy worker

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
Add new "true constraint" on strategy worker to remove need for QBI.
Add self constraint on GSM worker to remove last QBI query. 

The GSM was previously getting interest in the `SKELETON_ENTITY_MANIFEST_COMPONENT_ID` entities via a component constraint on the GSM entity. This has been reworked to give the GSM worker authority over a new tag component we add to the skeleton entity manifest entities, and then uses self-interest to gain interest in the rest of the entity. 

I'll leave the QBI queries within the snapshot entities for now, until we fully shift to USI.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SkeletonEntityManifestPublisher.cpp [hunk 1]
```diff
@@ -49,23 +49,27 @@ FManifestCreationHandle FSkeletonManifestPublisher::CreateManifestForPartition(I
...
+		Authority.Delegations.Emplace(SpatialConstants::SKELETON_ENTITY_MANIFEST_GSM_AUTH_COMPONENT_SET_ID,
+									  SpatialConstants::INITIAL_SNAPSHOT_PARTITION_ENTITY_ID);
 
 		SkeletonEntityManifestComponents.Emplace(
 			ComponentData(OwningComponentDataPtr(Authority.CreateComponentData().schema_type), AuthorityDelegation::ComponentId));
 	}
 
 	{
+		// Give GSM worker interest in this entity
+		SkeletonEntityManifestComponents.Emplace(ComponentData(SpatialConstants::SKELETON_ENTITY_MANIFEST_GSM_TAG_COMPONENT_ID));
```
Comment (jacquese) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SkeletonEntityManifestPublisher.cpp:61]: How does this give interest? There aren't component constraints here right?

**Review by jacquese — APPROVED** [review]
generally lgtm although I'm not sure I understood where the replacement USI queries for the previous unreal server QBI is defined here

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SkeletonEntityManifestPublisher.cpp (hunk 1 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SkeletonEntityManifestPublisher.cpp:61]: No - we're using self-interest here. GSM worker has auth over SKELETON_ENTITY_MANIFEST_GSM_TAG_COMPONENT_ID and then interest in SKELETON_ENTITY_MANIFEST_COMPONENT_ID

**Comment by m-samiec** [comment]
> generally lgtm although I'm not sure I understood where the replacement USI queries for the previous unreal server QBI is defined here

Updated PR description to detail what's going on in more detail. Let me know if you want further clarification.

**Review by improbable-dmitrii — COMMENTED** [review]
The actor system checks worry me, otherwise looks good.

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialStrategySystem.cpp [hunk 2]
```diff
@@ -345,7 +354,7 @@ void FSpatialStrategySystem::UpdateStrategySystemInterest(ISpatialOSWorker& Conn
 	{
 		Query ServerQuery = {};
 		ServerQuery.ResultComponentIds = { SpatialConstants::SERVER_WORKER_COMPONENT_ID };
-		for (auto& Component : ServerWorkerDataStorages.GetComponentsToWatch())
+		for (const Worker_ComponentId& Component : ServerWorkerDataStorages.GetComponentsToWatch())
```
[the remark this answers is not in the document]
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialStrategySystem.cpp:357]: No not necessary - I'd just replaced the auto here - but can remove the reference also.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp [hunk 3]
```diff
@@ -316,7 +316,7 @@ bool USpatialActorChannel::CleanUp(const bool bForDestroy, EChannelCloseReason C
 {
 	ValidateChannelNotBroken();
 
-	if (NetDriver != nullptr)
+	if (NetDriver != nullptr && NetDriver->ActorSystem != nullptr)
```
[the remark this answers is not in the document]
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp:319]: Hmm, in debugging this, I hit lots of edge cases in the startup flow. Specifically, ActorSystem doesn't get initialised until after the connection to spatial is established, but by that point the actor's are already spawned. Although reviewing the flow, they still shouldn't have had actor channels spawned for them (as that should be after begin play). I'm not sure it hurts to check the actor system though, as if it is invalid, we'll crash on some subsequent lines when attempting to dereference it. Thoughts?

**Pull request merged**